server/controllers/finance/fiscal.js

```diff
@@ -187,17 +210,250 @@ function update(req, res, next) {
+    .addQuery(sqlDelFiscalYear, [id])
+    .execute()
+    .then(results => {
+      // results[0] is the result for the first query
+      // results[1] is the result for the second query
+      if (!results[1].affectedRows) {
         throw new NotFound(`Cannot find fiscal year with id: ${id}`);
       }
-
       res.sendStatus(204);
     })
     .catch(next)
     .done();
 }
+
+/**
+ * Get /fiscal/:id/balance/:period
+ * @param {number} id the fiscal year id
+ * @param {number} period the period number [0,13]
+ * The balance for a specified fiscal year and period with all accounts
+ * the period must be given
+ */
+function getBalance(req, res, next) {
+  const id = req.params.id;
+  const period = req.params.period_number;
+
+  lookupBalance(id, period)
+  .then(rows => {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+ * @function lookupBalance
+ * @param {number} fiscalYearId fiscal year id
+ * @param {number} periodNumber the period number
+ */
+function lookupBalance(fiscalYearId, periodNumber) {
+  let glb = {};
+
+  const sql = `
+    SELECT t.period_id, a.id, a.label, t.debit, t.credit
+    FROM period_total t
+    JOIN account a ON a.id = t.account_id
+    JOIN period p ON p.id = t.period_id
+    WHERE t.fiscal_year_id = ? AND p.number = ?;
+  `;
+
+  const periodSql = `
+    SELECT id FROM period
+    WHERE fiscal_year_id = ? AND number = ?;
+  `;
+
+  return db.exec(periodSql, [fiscalYearId, periodNumber])
+  .then(rows => {
+    if (!rows.length) {
+      throw new NotFound(`Could not find the period ${periodNumber} for the fiscal year with id ${fiscalYearId}.`);
+    }
+    glb.period = rows[0];
+    return db.exec(sql, [fiscalYearId, periodNumber]);
+  })
+  .then(rows => {
+    glb.existTotalAccount = rows;
+
+    return AccountList;
+  })
+  .then(rows => {
+    let inlineAccount;
+
+    glb.totalAccount = rows.map(item => {
+      inlineAccount = _.find(glb.existTotalAccount, { id: item.id });
+
+      if (inlineAccount) {
+        item.period_id = inlineAccount.period_id;
+        item.debit = inlineAccount.debit;
+        item.credit = inlineAccount.credit;
+      } else {
+        item.period_id = glb.period.id;
+        item.debit = 0;
+        item.credit = 0;
+      }
+
+      return item;
+    });
+
+    return glb.totalAccount;
+  });
+}
+
+/**
+ * PUT /fiscal/:id/opening_balance
+ * set the opening balance for a specified fiscal year
+ */
+function setOpeningBalance(req, res, next) {
+  const id = req.params.id;
+  const fiscalYear = req.body.params.fiscal;
+  let accounts = req.body.params.accounts;
+
+  // check for previous fiscal year
+  hasPreviousFiscalYear(id)
+  .then(hasPrevious => {
+
+    if (hasPrevious) {
+      // load from the period N+1 into period 0
+      return loadOpeningBalance(fiscalYear);
+
+    } else {
+      // set new opening balance
+      return newOpeningBalance(fiscalYear, accounts);
+
+    }
+
+  })
+  .then(rows => res.status(201).send())
```

Comment:
Anytime you use `status(number).send()`, you should use `sendStatus(number)` instead.  [Here is the documentation](https://expressjs.com/en/4x/api.html#res.sendStatus).